### DISCUSSION TITLE
I Need guidelines for installing pimcore with bitnami and configure everyhing including Data Importer

### DISCUSSION THREAD

**Discussion by Abdalmohsen-H** [pimcore/pimcore#14121] [General]
Hi all, I'm struggling to find my way to make Data Importer working properly , I am using bitnami pimcore stack on debian 11.
My first try when follow the official installation guidelines I got an error on the first command 
 composer require pimcore/data-importer --with-all-dependencies

In RequireCommand.php line 142:
                                                                                
  file_put_contents(./composer.json): Failed to open stream: Permission denied  

So I add "sudo" before this command and I got warning 
Do not run Composer as root/super user! See https://getcomposer.org/root for details

I ignored this and continued back then and added the crontab commands then plugin installed put get stuck at 0/x proceed for ever when I get back to command line I found that if I tested cronjob commands outside crontab it fires an error

Could not open input file: /opt/bitnami/stack/pimcore/bin/console
and so on .

So I stopped this aws instance and creted fresh new one

Now I need your help to install  Data Importer again but properly this time and add all needed configurations for bitnami pimcore stack version 10.5.15.1 on debian 11

**Comment by AndrewKMoses** [comment]
I run composer with sudo and then restore owner of the files in pimcore directory to daemon:daemon.

**Reply by Abdalmohsen-H** [reply]
Thank you @AndrewKMoses 
Also a better option is to run commands as demon
